packages/@sanity/base/src/components/DevServerStatus.js

```diff
@@ -7,15 +7,16 @@ const EventSource = isWindowEventSource
   ? window.EventSource // Native browser EventSource
   : polyfilledEventSource // Node.js, IE, Edge etc
 
-const STATE_CLOSED = 0
+const STATE_CONNECTING = 0
 const STATE_OPEN = 1
+const STATE_CLOSED = 2
 
 class DevServerStatus extends PureComponent {
   constructor(...args) {
     super(...args)
     this.enabled = __DEV__ && EventSource
     this.state = {
-      connectionState: STATE_CLOSED,
+      connectionState: STATE_OPEN,
```

Comment:
Was this intentional? Until the eventsource is actually open, it should probably stay as closed? :)
If this is to prevent an initial "we are disconnected", I think we should use the `hasHadConnection` flag instead:

Only show the disconnected message if we have previously been connected and we are now in closed or reconnecting state.

Reply:
It's to prevent the snackbar showing for a split second when the Studio starts! :/

Reply:
I think my hasHadConnection proposal would do the trick, then?

Reply:
Alright! Will test tomorrow, thank you! 😃